Modules/Sources/Storage/CoreData/CoreDataIterativeMigrator.swift

```diff
@@ -176,4 +187,37 @@ private extension CoreDataIterativeMigrator {
             .appendingPathComponent("migration_\(UUID().uuidString)")
             .appendingPathExtension("sqlite")
     }
+
+    func shouldDestroyPersistentStore(for sourceModel: NSManagedObjectModel) -> Bool {
+        guard let sourceVersion = CoreDataMigratorUtils.findSourceVersion(for: sourceModel, in: modelsInventory) else {
+            DDLogInfo("Could not determine source model version. Using iterative migration.")
+            return false
+        }
+        // If the merchant is on a model version older than this, nuke the DB and migrate them to the latest one without iterative process
+        let oldestModelThreshold = Constants.oldestSupportedDataModel
+        let versionNumber = CoreDataMigratorUtils.extractVersionNumber(from: sourceVersion.name)
```

Comment:
nit: how about combining `CoreDataMigratorUtils.findSourceVersion` and `CoreDataMigratorUtils.extractVersionNumber` into one function to simplify the code a bit? It looks like the two methods are only called here.

```suggestion
        guard let versionNumber = CoreDataMigratorUtils.findSourceVersionNumber(for: sourceModel, in: modelsInventory) else {
            DDLogInfo("Could not determine source model version. Using iterative migration.")
            return false
        }
        // If the merchant is on a model version older than this, nuke the DB and migrate them to the latest one without iterative process
        let oldestModelThreshold = Constants.oldestSupportedDataModel
```

Reply:
In general I'd prefer to keep these separated, but with the simplification you suggested on https://github.com/woocommerce/woocommerce-ios/pull/15987/commits/608096441ab9aa77e9f5a0e71952487e9ff0318d we can also remove `extractVersionNumber` entirely.